src/libraries/Microsoft.PowerFx.Connectors/ODataVisitor.cs

```diff
@@ -0,0 +1,237 @@
+            try
+            {
+                FormulaValue result = runContext.EvalAsync(node).Result;
+                return SerializeLiteralValue(result);
+            }
+            catch (KeyNotFoundException)
+            {
+                // scope access to table value
+                throw new NotDelegableException();
+            }
+        }
+
+        public override string Visit(BinaryOpNode node, DelegationRunContext runContext)
+        {
+            return $"{node.Left.Accept(this, runContext)} {ODataOpFromBinaryOp(node.Op)} {node.Right.Accept(this, runContext)}";
+        }
+
+        public override string Visit(UnaryOpNode node, DelegationRunContext runContext)
+        {
+            return $"{ODataOpFromUnaryOp(node.Op)}({node.Child.Accept(this, runContext)})";
+        }
+
+        public override string Visit(ScopeAccessNode node, DelegationRunContext runContext)
+        {
+            if (node.Value is ScopeAccessSymbol symbol)
+            {
+                return symbol.Name.Value;
+            }
+
+            return string.Empty;
+        }
+
+        public override string Visit(RecordFieldAccessNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        public override string Visit(ResolvedObjectNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        public override string Visit(SingleColumnTableAccessNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        public override string Visit(ChainingNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        public override string Visit(AggregateCoercionNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        private static string SerializeLiteralValue(FormulaValue value)
+        {
+            return value switch
+            {
+                BooleanValue booleanValue => booleanValue.Value ? "true" : "false",
+                ColorValue colorValue => throw new NotDelegableException(),
+                DateTimeValue dateTimeValue => dateTimeValue.Value.ToString("yyyy-MM-ddTHH:mm:ssZ"),
+                DateValue dateValue => dateValue.Value.ToString("yyyy-MM-dd"),
+                GuidValue guidValue => guidValue.Value.ToString(),
+                NumberValue numberValue => numberValue.Value.ToString(CultureInfo.InvariantCulture),
+                StringValue stringValue => $"'{stringValue.Value}'",
+                TimeValue timeValue => throw new NotDelegableException(),
+                _ => throw new NotDelegableException(),
+            };
+        }
+
+        private static string ODataOpFromUnaryOp(UnaryOpKind op) =>
+            op switch
+            {
+                UnaryOpKind.Negate => "NOT",
```

Comment:
same here. Also, can this be consolidated with the Not Function Call impl?

Reply:
consolidated